.github/workflows/publish-site.yml

```diff
@@ -0,0 +1,41 @@
+name: Publish Site
+on:
+  push:
+      branches:
+        - master
+      paths:
+        - 'README.md'
+        - 'site/default.html'
+        - 'site/generate.site.mjs'
+
+jobs:
+  main:
+    name: Publish Site
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: write
+      contents: read
+
+    steps:
+      - uses: linz/action-typescript@v3
+
+      - name: AWS Configure
+        uses: aws-actions/configure-aws-credentials@v2
+        with:
+          aws-region: ap-southeast-2
+          mask-aws-account-id: true
+          role-to-assume: ${{ secrets.AWS_CI_ROLE }}
+
+      - name: AWS Configure ODR
+        uses: aws-actions/configure-aws-credentials@v2
+        with:
+          aws-region: ap-southeast-2
+          mask-aws-account-id: true
+          role-to-assume: ${{ secrets.AWS_ODR_DATAMANAGER_ROLE }}
+          role-chaining: true
+
+      - name: Deploy page to S3
+        run: |
+          ./site/get.assets.sh
+          node site/generate.site.mjs
```

Comment:
in terms of saftey, this should be run before we login to AWS, as this script would now have permissions to deploy assets into this bucket.